### ISSUE TITLE
Form Data Validation

### ISSUE THREAD

**Issue by mpk3** [mpk3/DjangoWebApps#4]
- [x] There is an issue where if users submit the wrong datatypes the webpage will 404
- [x] Create form class in django
- [x] Connect form class to html
- [x] Meta values in models should be adjusted to also not allowed illegal datatypes
- [x] Use validators
Checkout the DjangoTDD Forms section parts 13 and 14 for reference

- [x] Fix form formatting so delete button is included; implement multiple form actions pg 135 _Django Design Patterns and Best Practices_ (Packt)(2018)

**Comment by mpk3** [comment]
I connected and created the form today. Going forward whenever I need to use forms I should start with a django form from the beginning. The first version of the site used a form I manually wrote in the html. I also spent a bit of time doing HTML/CSS formatting so that the page looked okay. It then took me a day-ish to go through and wire the form so that it passed all the tests. I think some time was wasted on dealing with hooking up the form and formatting the html because all of it had to be kind of re-done. It was by no means rewritten entirely. Frankly TDD really made everything going relatively well. BUT, going forward I think best practices would be to use Django Forms from the beginning. They are actually easier anways.
